libraries/chain/include/steem/chain/smt_objects/account_balance_object.hpp

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <steem/chain/steem_object_types.hpp>
+#include <steem/protocol/smt_operations.hpp>
+
+#ifdef STEEM_ENABLE_SMT
+
+namespace steem { namespace chain {
+
+/**
+ * Class responsible for holding non STEEM balance (SMT one) for given account.
+ * Since multiple kinds of balances are possible (atm regular and savings) this
+ * class is parameterized to distinguish their types. 
+ */
+template <enum object_type ObjectType>
+class account_balance_object : public object< ObjectType, account_balance_object<ObjectType> >
+{
+   account_balance_object() = delete;
+
+public:   
+   typedef account_balance_object<ObjectType>   _this_type;
+   using typename object< ObjectType, _this_type >::id_type;
+
+   template <class Constructor, class Allocator>
+   account_balance_object(Constructor&&, allocator< Allocator >)
+   {
+      c( *this );
+   }
+
+   // id_type is actually oid<account_balance_object<ObjectType> >
+   id_type             id;
+   /// Name of the account, the balance is held for.
+   account_name_type   owner = "@@@@@";
```

Comment:
I am not sure why we are setting the default value for account name type when the default `account_name_type()` is already an invalid account name.

Reply:
Looks like this happens in other places, we can file an issue to fix it.

Reply:
It isn't wrong, per se, but is not needed and does not match other places where account names are present in objects. I think this was picked up because some of our out of source documentation uses it as a place holder for an arbitrary account.